bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/storage/DisposableStorage.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.storage;
+
+/**
+ * A {@code Storage} that could be disposed.
+ *
+ * @author Markus Rathgeb - Initial Contribution and API
+ */
+public interface DisposableStorage<T> extends Storage<T> {
```

Comment:
I'm not 100% sure whether this is the best name, because (at least to me) it is associated destroying UI elements /freeing memory etc. and not with something as dangerous and terminal as deleting the database. How about `DeletableStoryge` - weird word, not happy with this suggestion myself I have to admit though 😕